www/js/diary/list/TimelineScrollList.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { FlashList } from '@shopify/flash-list';
+import { array, func, object, oneOfType, bool, string } from 'prop-types';
+import TripCard from '../cards/TripCard';
+import PlaceCard from '../cards/PlaceCard';
+import UntrackedTimeCard from '../cards/UntrackedTimeCard';
+import { View } from 'react-native';
+import { ActivityIndicator } from 'react-native-paper';
+import LoadMoreButton from './LoadMoreButton';
+
+const renderCard = ({ item: listEntry }) => {
+  if (listEntry.origin_key.includes('trip')) {
+    return <TripCard trip={listEntry} />
+  } else if (listEntry.origin_key.includes('place')) {
+    return <PlaceCard place={listEntry} />
+  } else if (listEntry.origin_key.includes('untracked')) {
+    return <UntrackedTimeCard triplike={listEntry} />
+  }
+};
+
+const separator = () => <View style={{ height: 8 }} />
+const bigSpinner = <ActivityIndicator size="large" style={{margin: 15}} />
+const smallSpinner = <ActivityIndicator size="small" style={{margin: 5}} />
+
+const TimelineScrollList = ({ listEntries, loadedRange, pipelineRange, loadMoreFn, isLoading }) => {
+
+  // The way that FlashList inverts the scroll view means we have to reverse the order of items too
+  const reversedListEntries = listEntries ? [...listEntries].reverse() : [];
```

Comment:
would it be more performant to maintain the list in reversed order in the `LabelTab`?

Reply:
Doesn't the server give us oldest -> newest?

We could do the reversal sooner if that's what you're asking. But I think the reversal has to happen at some point?

Reply:
Ah, I see how it would make a difference for loading subsequent batches into view
*TODO later*

Reply:
I still want to consider this, but I think it is lower priority and can be deferred until a later cleanup